clippy_dev/src/lint.rs

```diff
@@ -12,14 +12,26 @@ pub fn run<'a>(path: &str, args: impl Iterator<Item = &'a String>) {
     };
 
     if is_file {
+        // Support setting edition by passing `--edition 20XX` as arg.
```

Comment:
Can we let clap parse this in main.rs or is this intentionally done in this way?

Reply:
args for `cargo dev lint` are not parsed by clap as you can parse arbitrary ones, and the `--edition 20XX` is only available when you pass it a file and not a dir, so I don't think we can do it with clap.

Reply:
Indeed, this is much better.